### PR TITLE
Implementing the requested example

### LINKED ISSUE #1381: Create a code example for category prediction [$15]
See http://lists.numenta.org/pipermail/nupic_lists.numenta.org/2014-June/004057.html

### PR DESCRIPTION
**Pull request by passiweinberger**
this is the code by @oxtopus a bit modyfied and added as a example

@oxtopus : please let me know what charity you would like to money to go to :)

fixes #1381

### CODE REVIEW

**Comment by passiweinberger** [comment]
addresses #1381

File: examples/prediction/category_prediction/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Run with ' ./run.py'
+
+clean directory with './clean.py'
+
+change the stopwords in stopwords.txt if wanted
+
+You can also choose a different story to use. For JACK AND THE BEANSTALK use indicies 129017 - 138889.
+
+After running the model, tokens.txt will contain all the relevant word tokens and results.csv will list the predicted probabilities for each.
```
Comment (breznak) [inline examples/prediction/category_prediction/README.md]: can you add a high-level description what this example does?

File: examples/prediction/category_prediction/run.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        outfile.write(token + '\n')
+    if inp.tell() >= 138889:
+      break
+inp.close()
+outfile.close()
+
+
+# Create and run the model:
+
+MODEL_PARAMS = {
```
Comment (breznak) [inline examples/prediction/category_prediction/run.py]: if you want, you could also add swarming to the example

**Comment by rhyolight** [comment]
I added the issue link in the PR description.

**Comment by breznak** [comment]
@passiweinberger looks good to me :+1: just added a few suggestions, could you also mention this example in our wiki about "Classification with NuPIC" (we have something like that)

**Comment by oxtopus** [comment]
Thanks @passiweinberger!  Please donate to [Girls Who Code](http://girlswhocode.com/)

File: examples/prediction/category_prediction/run.py (hunk 2 above)
Comment (passiweinberger) [inline examples/prediction/category_prediction/run.py]: thought about it :) maybe I will... tomorrow :P just wanted to get it done ;)

**Comment by passiweinberger** [comment]
@breznak  done. but a dead link until its merged ;)

**Comment by breznak** [comment]
:+1: thanks!

**Comment by passiweinberger** [comment]
is this OK ?

**Comment by rhyolight** [comment]
@breznak said:
> could you also mention this example in our wiki about "Classification with NuPIC"

This? https://github.com/numenta/nupic/wiki/Spatial-Classification

**Comment by breznak** [comment]
>> could you also mention this example in our wiki about "Classification with NuPIC"

> This? https://github.com/numenta/nupic/wiki/Spatial-Classification

@rhyolight  yes

**Comment by passiweinberger** [comment]
@rhyolight Oh sry, do you want me to remove the ones in the general wiki and add them there, or just add it there and also keep it in the general wiki under code-examples ?

**Comment by breznak** [comment]
also consider placing the example here instead, `examples/opf/experiments/classification/` ?

**Comment by oxtopus** [comment]
@breznak It's an example demonstrating basic setup of model configured with CategoryEncoder.  The goal is to have something that someone can follow and adapt for their own purposes.  The specific application is somewhat trivial: it predicts the next word for each word in a story, treating each word as a predefined category.

**Comment by passiweinberger** [comment]
i find classification_prediction more fitting for this ^^ as it more predicts the next word than classifies them... as far as i observed it. @oxtopus might be better to specify.

**Comment by passiweinberger** [comment]
Ups... stupic mobile version :-1:

**Comment by breznak** [comment]
> i find classification_prediction more fitting for this

yep, I got my high-level description, so now I understand and agree. Sorry for the hassle :)

**Comment by passiweinberger** [comment]
What's missing? :)

**Comment by rhyolight** [comment]
Oh, nothing now. Looks like it was fixed.

**Comment by rhyolight** [comment]
@passiweinberger I cleaned up some stuff. Merge https://github.com/passiweinberger/nupic/pull/1 if you like.

**Comment by rhyolight** [comment]
@passiweinberger Thanks!

**Comment by passiweinberger** [comment]
:) You're welcome. Let me know if there is anything  (more basic stuff) needed. I'd love to do more, but am not yet too comfortable coding in general so pass 'easy' but challenging stuff to me. This, I think is one of the best ways to get my HTM trained ;)

**Comment by rhyolight** [comment]
Ok, here are a bunch of tasks labeled "newbie":
https://github.com/numenta/nupic/labels/newbie

---------
Matt Taylor
OS Community Flag-Bearer
Numenta

**Comment by zmh0510** [comment]
@rhyolight  HI! After I run the run.py, i got the results.csv. But the result confused me. It shows the predicted next word for each word is the previous word exactly.  I don't think it achieves the prediction task.  Can you explain it please? thx

**Comment by rhyolight** [comment]
Looks like something is broken to me. https://github.com/numenta/nupic/issues/3790

**Pull request merged**